feat: Expose ARGOCD_APP_NAME to the `kustomize build` command (#8098)

wd opened the pull request:
See the discussion at here: https://github.com/argoproj/argo-cd/issues/8098

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

wd commented:
It looks like the CI job needs to approve after every commit? ..

alexmt commented:
>It looks like the CI job needs to approve after every commit? ..

 Correct, Github requires it for the first-time contributors. Just approved it one more time.

alexmt commented on docs/user-guide/build-environment.md:
```diff
@@ -11,4 +11,10 @@
\ No newline at end of file
+* `KUBE_API_VERSIONS` = the version of kubernetes API
+
+> v2.3.0
```
According to changes in `repository.go` kustomize is going to receive the same environment variables as other tools. I think this is correct. Can you please update the document accordingly?

alexmt replied:
After migrating to https://readthedocs.io/ we no longer need to specify versions documentation ( `v1.4` , `v2.3.0` etc ) . Can you remove versions in this document please?

alexmt commented on reposerver/repository/repository.go:
```diff
@@ -1426,7 +1426,10 @@ func populateKustomizeAppDetails(res *apiclient.RepoAppDetailsResponse, q *apicl
 		kustomizeBinary = q.KustomizeOptions.BinaryPath
 	}
 	k := kustomize.NewKustomizeApp(appPath, q.Repo.GetGitCreds(), q.Repo.Repo, kustomizeBinary)
-	_, images, err := k.Build(q.Source.Kustomize, q.KustomizeOptions)
```
I think kustomize should receive the same env variables as all other tools. Please use `newEnv` instead of creating env with one variable.

wd replied:
Thank you for your review.

Yeah. I know I should use the `newEnv` defined [here](https://github.com/argoproj/argo-cd/blob/a17c77e238f44b442020e3779b8284879f5fb7f3/reposerver/repository/repository.go#L835). However it needs to pass the `apiclient.ManifestRequest` as the param, and there is no such variable in this [context](https://github.com/argoproj/argo-cd/blob/dfd6ec8e08a83f34259a9620a17761991e0fe711/reposerver/repository/repository.go#L1422-L1438). I don't want to change to much code at the first contribution. I thought maybe adding the `ARGOCD_APP_NAME` is also acceptable. 
Anyway, I can investigate more if you'd prefer to use the `newEnv` here also.

wd replied:
It looks like I can construct the `ManifestRequest` with most of the necessary properties by myself without the `Namespace`. I don't know where to find the destination application namespace.
```
	mq := apiclient.ManifestRequest{
		AppName: q.AppName,
		Namespace: "",
		Repo: q.Repo,
		ApplicationSource: q.Source,
 	}
```

alexmt replied:
This approach looks good. I would omit namespace for now. We can add the namespace to API in a separate PR

wd commented:
@alexmt I have changed the code as you said, could you take a look again?

changes I have made
1. remove the version number from the doc about the environment vars. added kustomize to the list.
2. use the `newEnv` to generate the environment variables as other tools. 
3. leave a `FIXME` mark around the `namespace` varialbe.

alexmt reviewed (approved):
LGTM. Thank you!
